subprojects/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/tasks/ValidateTaskPropertiesIntegrationTest.groovy

```diff
@@ -387,6 +391,52 @@ class ValidateTaskPropertiesIntegrationTest extends AbstractIntegrationSpec {
         file("build/reports/task-properties/report.txt").text == ""
     }
 
+    @Unroll
+    def "report setters for property with mutable type #type"() {
+        @Language("JAVA") myTask = """
+            import org.gradle.api.DefaultTask;
+            import org.gradle.api.tasks.InputFiles;
+
+            public class MyTask extends DefaultTask {
+                // no field, just getter and setter
+                @InputFiles public ${type} getMutablePropertyWithSetter() { return null; } 
+                public void setMutablePropertyWithSetter(${type} value) {} 
+
+                // no field, just getters
+                @InputFiles public ${type} getMutablePropertyWithoutSetter() { return null; } 
+
+                // no field, just setter
+                public void setMutablePropertyWithoutGetter() {} 
+
+                // field with getter and setter
+                private ${type} fieldWithGetterAndSetter;                
+                @InputFiles public ${type} getFieldWithGetterAndSetter() { return null; } 
+                public void setFieldWithGetterAndSetter(${type} value) {} 
+                
+                // final field with constructor and getter
+                private final ${type} finalFieldWithoutSetter; 
+                MyTask() { finalFieldWithoutSetter = null; } 
+                @InputFiles public ${type} getSources() { return null; } 
+                
+            }
+        """
+        file("src/main/java/MyTask.java") << myTask
+
+        when:
+        fails "validateTaskProperties"
+
+        then:
+        def simpleType = type.replaceAll("<.+>", "")
+        def lines = file("build/reports/task-properties/report.txt").readLines()
+        lines.toSet() == [
```

Comment:
seems more robust to use sets here, instead of exact string comparison

Reply:
There's no need for this verbose logic here, as the output in the file is sorted. Take a look at how the other test cases check the results:

```groovy

        file("build/reports/task-properties/report.txt").text == """
            Warning: Type 'MyTask': property 'badTime' is not annotated with an input or output annotation.
            Warning: Type 'MyTask': property 'oldThing' is not annotated with an input or output annotation.
            Warning: Type 'MyTask': property 'options.badNested' is not annotated with an input or output annotation.
            Warning: Type 'MyTask': property 'ter' is not annotated with an input or output annotation.
        """.stripIndent().trim()
```

Reply:
this is what I wanted to avoid, but I'll revert